src/ProjectOrigin.VerifiableEventStore/Services/Repository/ITransactionRepository.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ProjectOrigin.VerifiableEventStore.Models;
+
+namespace ProjectOrigin.VerifiableEventStore.Services.Repository;
+
+public interface ITransactionRepository
+{
+    Task Store(StreamTransaction @event);
+
+    Task<ImmutableLog.V1.Block?> GetBlockFromTransactionHash(TransactionHash transactionHash);
+    Task<TransactionStatus> GetTransactionStatus(TransactionHash transactionHash);
+    Task<IList<StreamTransaction>> GetStreamTransactionsForBlock(BlockHash blockHash);
+    Task<IList<StreamTransaction>> GetStreamTransactionsForStream(Guid streamId);
+
+    Task<NewBlock?> CreateNextBlock();
+    Task FinalizeBlock(BlockHash hash, ImmutableLog.V1.BlockPublication publication);
+
+    public record NewBlock(ImmutableLog.V1.BlockHeader Header, IList<TransactionHash> TransactionHashes);
```

Comment:
Is it on purpose that this record is defined within the interface?

Reply:
kinda, in the first iteration i returned a tuple of the two variables, but needed to be able to return a null reference when no new block could be created because of no transactions to put in the block.. But it could also have been achived using a try pattern

```csharp
bool TryCreateNextBlock(out ImmutableLog.V1.BlockHeader Header, out IList<TransactionHash> TransactionHashes);
```

This might actually make it more readable, but it always depends on people preference of the try methods with multiple out parameters.